pandas/_libs/parsers.pyx

```diff
@@ -454,7 +454,8 @@ cdef class TextReader:
             if callable(usecols):
                 self.usecols = usecols
             else:
-                self.usecols = set(usecols)
+                self.usecols = list(set(usecols))
+                self.usecols.sort()
```

Comment:
Is there an existing test that highlights where this matters?

Reply:
sorry, this commit is bogus. I'm not sure why but somehow this changes the dtype in one of the subtests in ``TestCParserHighMemory.test_usecols_with_parse_dates_and_usecol_names`` on PyPy, col 'a' gets the correct ``datetime64[ns]`` when usercols is [0, 2, 3] but ``object`` dtype when usercols is [3, 2, 0]. Something else is going on, I will back this out and look for the real culprit.